orderapi/src/main/java/fi/hel/verkkokauppa/order/service/accounting/AccountingSlipService.java

```diff
@@ -272,6 +404,43 @@ private List<AccountingSlipRowDto> createAccountingSlipRowDtos(List<OrderItemAcc
                     .wbsElement(summedItemAccounting.getProject())
                     .functionalArea(summedItemAccounting.getOperationArea())
                     .balanceProfitCenter(summedItemAccounting.getBalanceProfitCenter())
+                    .rowType(AccountingRowTypeEnum.ORDER)
+                    .build();
+
+            rows.add(accountingSlipRowDto);
+            rowNumber++;
+        }
+
+        return rows;
+    }
+
+    private List<AccountingSlipRowDto> createAccountingSlipRowDtosForRefunds(List<RefundItemAccountingDto> summedItemAccountings, String accountingSlipId,
+                                                                             String lineText, int rowCountOffset) {
+        List<AccountingSlipRowDto> rows = new ArrayList<>();
+        if (summedItemAccountings == null) {
+            return rows;
+        }
+
+        int rowNumber = 1 + rowCountOffset;
+
+        for (RefundItemAccountingDto summedItemAccounting : summedItemAccountings) {
+            String accountingSlipRowId = UUIDGenerator.generateType3UUIDString(accountingSlipId, Integer.toString(rowNumber));
+
+            AccountingSlipRowDto accountingSlipRowDto = AccountingSlipRowDto.builder()
+                    .accountingSlipRowId(accountingSlipRowId)
+                    .accountingSlipId(accountingSlipId)
+                    .taxCode(summedItemAccounting.getVatCode())
+                    .amountInDocumentCurrency(formatSum(summedItemAccounting.getPriceGrossAsDouble()))
+                    .baseAmount("+" + formatSum(summedItemAccounting.getPriceNetAsDouble()))
```

Comment:
Is that + correct when using refunds?

Reply:
for orders there is automatically - for sum rows and + added for balance rows (always + or - before numbers). For refunds I added + for sum rows and there is automatically - for balance rows. Jarkko approved the example xml.